### DISCUSSION TITLE
Itemcolor (Chips) not gray

### DISCUSSION THREAD

**Discussion by kroessig** [Blackymas/NSPanel_HA_Blueprint#978] [Ideas]
Hey, 

i can choose a defined iconcolor for chips in the blueprint. 
Today, my heating was activated and for the first time i saw a red (orange) icon instead of the geay temperature icon. 

I change my icons dependend on a entity, --> that's working!  but i'm not able to chage the icon color.  
For example, i created a switch:Hey, 

i can choose a defined iconcolor for chips in the blueprint. 
Today, my heating was activated and for the first time i saw a red (orange) icon instead of the geay temperature icon. 

I change my icons dependend on a entity, --> that's work!  but i'm not able to chage the icon color.  
For example, i created a switch:

```
  - platform: template
    switches:
      waermepumpenstatussymbol_switch:
        unique_id: waermepumpenstatussymbol_switch
        value_template: "{{ is_state('input_boolean.waermepumpenstatussymbol', 'on') }}"
        turn_on:
          service: switch.turn_on
          target:
            entity_id: input_boolean.waermepumpenstatussymbol
        turn_off:
          service: switch.turn_off
          target:
            entity_id: input_boolean.waermepumpenstatussymbol
        icon_template: >-
          {% if is_state('sensor.luxtronik2_status','no request')  %}
          mdi:hvac-off
          {% elif is_state('sensor.luxtronik2_status','heating') %}  
          mdi:heating-coil
          {% elif is_state('sensor.luxtronik2_status','hot water') %}  
          mdi:fire
          {% else %}
          mdi:help
          {% endif %} 
```
 First:
Can someone tell me how to change the item color in my configuration?

Second:
Is it possible to hand over the icon-color to the nspanel?  I think in the blueprint a default-color is already set  (R128 G128 B128) ? 
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/139384422/72a4e28b-893e-4795-abca-f8a99cfcf5fd)

Thanks a lot!!!

**Comment by edwardtfn** [comment]
> First:
> Can someone tell me how to change the item color in my configuration?

Unfortunately neither Home Assistant or this Blueprint have support for a template icon color, however, on the coming version, the chip for the main climate (internal thermostat) entity (and only that one) will follow the standard colors from Home Assistant.

> Second:
> Is it possible to hand over the icon-color to the nspanel? I think in the blueprint a default-color is already set (R128 G128 B128) ?

Could you plea2e clarify your question? The color you pick on the Blueprint settings should be reflected on the icon visualization on your panel. Isn't that working?

**Comment by kroessig** [comment]
hey Edward,

thanks for the response.,

My 2nd question based on the first one. If there is no solution, the 2nd question is "obsolete".
The chip-icon are "nice" but small. A colored-icon (like the thermostat icon  - depended on a entiity state) would be my wish.

Therefor a colored icon in HA would be a good base. But if this is no standard,  it is complex for a repository (automatisation) like this. 
I changed from OpenHab to HomeAssistant last week, so i'm not firm in all regulations.

**Reply by edwardtfn** [reply]
Which integration are you using for your thermostat system? There's no climate entity available?
Using a climate entity will take care of all the icon selection you are looking for.

For the icon size, it's a bit more difficult as the (Nextion) display used by NSPanel doesn't support dunamically resize, so that size is hard coded.

**Reply by kroessig** [reply]
I use [https://github.com/BenPru/luxtronik](https://github.com/BenPru/luxtronik).
I works fine, also with this blueprint. The climate-icon is changing when the heat pump is running. All Fine. 

My wish / question:, is it possible to change the color of the other chips 1-7 dependend on a  specific endity state? I can change the icon with my code, but the color is gray (standard) oder fixed with the blueprint.
